### ISSUE TITLE
Dropdown isn't aligned properly when the focus is in the input field

### ISSUE THREAD

**Issue by antonmoiseev** [vaadin/vaadin-combo-box#201]
Steps to reproduce:
1. Asynchronously request the data
2. Put the cursor into the input field
3. When data arrives, initialize `items` property, invoke `open()` method to open the dropdown.

Result - the dropdown is not rendered next to the input field, items in the list are not visible.

Here is a JSFiddle to reproduce: https://jsfiddle.net/2ttsyzd2/2/
Here is a short video demoing the issue: https://www.dropbox.com/s/sf8atgdyymfq48a/issue.mp4?dl=0

**Comment by antonmoiseev** [comment]
Probably the issue is [here][src]:
```js
var overlayRect = this.getBoundingClientRect();
this._translateX = targetRect.left - overlayRect.left + (this._translateX || 0);
this._translateY = targetRect.top - overlayRect.top + (this._translateY || 0) + this._verticalOffset(overlayRect, targetRect);
this.translate3d(this._translateX + 'px', this._translateY + 'px', '0');
```
When the dropdown is empty all the dimensions of the `overlayRect` are `0` and I think `_translateX` and `_translateY` are not computed properly. As I understand this code you use `translate3d()` to move dropdown when it's already displayed to the user to avoid flickering. But when it's empty and is not rendered translating is not required, we can just display it right below the target (or above, depending on the `_alignedAbove`. Here is a quick fix I applied locally:
```diff
  var overlayRect = this.getBoundingClientRect();
+ if (overlayRect.width === 0 && overlayRect.height === 0) {
+   this.style.left =  targetRect.left;
+   this.style.top =  this._verticalOffset(overlayRect, targetRect);
+ } else {
    this._translateX = targetRect.left - overlayRect.left + (this._translateX || 0);
    this._translateY = targetRect.top - overlayRect.top + (this._translateY || 0) + this._verticalOffset(overlayRect, targetRect);
    this.translate3d(this._translateX + 'px', this._translateY + 'px', '0');
+ }
```
It solves the problem for me, but I wonder what do you think about it?


[src]: https://github.com/vaadin/vaadin-combo-box/blob/540c8aaa326a412c31e4a35b786fbff59097bf3a/vaadin-overlay-behavior.html#L92-L98

**Comment by tomivirkki** [comment]
Couldn't make the suggested change to work in fixing the issue. Made a different fix in 5e8c91b92f6be37584c7e7f790008788234ce3ae

@antonmoiseev can you verity that this fixes your original issue also?

**Comment by antonmoiseev** [comment]
Yes, thank you! Looking forward to get the PR merged.
